src/test/ui/coercion-missing-tail-expected-type.stderr

```diff
@@ -0,0 +1,19 @@
+error[E0308]: mismatched types
+  --> $DIR/coercion-missing-tail-expected-type.rs:13:28
+   |
+13 |   fn plus_one(x: i32) -> i32 {
+   |  ____________________________^
+14 | |     x + 1;
+15 | | }
+   | |_^ expected (), found i32
+   |
+   = note: expected type `i32`
```

Comment:
Good catch @arielb1. To fix that you have to change this line (part of `fn coerce_inner()`, not found in the diff):

```rust
fcx.eq_types(true, cause, expression_ty, self.merged_ty())
```

to be something like this:

```rust
fcx.eq_types(label_expression_as_expected, cause, expression_ty, self.merged_ty())
```

That `true` argument indicates that the `a` value (first type) ought to be marked as "expected". That is no longer *always* true.